src/include/ipc/topology.h

```diff
@@ -182,9 +182,16 @@ struct sof_ipc_comp_process {
 	struct sof_ipc_comp_config config;
 	uint32_t size;	/**< size of bespoke data section in bytes */
 	uint32_t type;	/**< sof_ipc_effect_type */
+	uint32_t min_sink_samples;   /**< min sink buffer size measured in
+				       *  samples required to run module
+				       */
+	uint32_t min_source_samples; /**< amount of data measured in
+				       *  samples available at source buffer
+				       *  required to run module
+				       */
```

Comment:
There will be cases where effects will take PCM data and generate a byte stream. Same for encoders.
Decoders are the other way around, you take as input byte data and generate samples.

Maybe instead of strong-typing these two parameters, we should have an additional flag that describes the input/output data type.

Reply:
@plbossart 
I've changed ```min_source/sink_samples``` to min_source/sink_bytes``` as samples could be easily converted to bytes.